pkg/errors/errors_test.go

```diff
@@ -0,0 +1,133 @@
+package errors
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestError_Error(t *testing.T) {
+	tests := []struct {
+		name     string
+		errorObj *Error
+		expected string
+	}{
+		{
+			name: "simple error with message only",
+			errorObj: &Error{
+				Code:    "123",
+				Message: "simple message",
+			},
+			expected: "simple message",
+		},
+		{
+			name: "error with parameters",
+			errorObj: &Error{
+				Code:    "123",
+				Message: "message with %s and %d",
+				Params:  []interface{}{"string", 42},
+			},
+			expected: "message with string and 42",
+		},
+		{
+			name: "error with nested error",
+			errorObj: &Error{
+				Code:    "123",
+				Message: "parent message",
+				Err:     errors.New("child error"),
+			},
+			expected: "parent message: child error",
+		},
+		{
+			name: "error with nested error and parameters",
+			errorObj: &Error{
+				Code:    "123",
+				Message: "message with %s",
+				Params:  []interface{}{"parameter"},
+				Err:     errors.New("child error"),
+			},
+			expected: "message with parameter: child error",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			assert.Equal(t, tt.expected, tt.errorObj.Error())
+		})
+	}
+}
+
+func TestError_Wrap(t *testing.T) {
+	tests := []struct {
+		name      string
+		errorObj  *Error
+		wrapError error
+		expected  string
+	}{
+		{
+			name: "wrap no error",
+			errorObj: &Error{
+				Message: "test message",
+				Err:     errors.New("initial error"),
+			},
+			wrapError: nil,
+			expected:  "test message: initial error",
+		},
+		{
+			name: "wrap an error",
+			errorObj: &Error{
+				Message: "test message",
+				Err:     errors.New("initial error"),
+			},
+			wrapError: errors.New("wrapped error"),
+			expected:  "test message: initial error\nwrapped error",
+		},
```

Comment:
One additional test case could be when the wrapError is a joined error itself. 
Also some opportunity for refactoring common errors
```go
initialError := errors.New("initial error")
wrapError1 := errors.New("wrapped error 1")
wrapError2 := errors.New("wrapped error 2")
...
		{
			name: "wrap a joined error",
			errorObj: &Error{
				Message: "test message",
				Err:     initialError,
			},
			wrapError: errors.Join(wrapError1, WrapError2),
			expected:  "test message: initial error\nwrapped error 1\nwrapped error 2",
		},
```

Reply:
added a test for it anyways